### ISSUE TITLE
Focus Management and Up and Down Cursor keys

### ISSUE THREAD

**Issue by terracoda** [phetsims/balloons-and-static-electricity#153]
Using the reader I noticed the following issues with the down cursor key and how descriptions are read out:
1. The label of the Grab Balloon button is not read out, and the button (the balloon) does not gain visual focus, whereas this is how the other buttons work.
2. The Yellow balloon description (before interaction), contains no position nor charge information. It only contains the navigation cue which is actually inoperable when the balloon is not grabbed.
3. When using the up cursor key (i.e. moving backwards), it would make sense to have the control gain focus when the user cursors to the control's description (is that possible?). Using the down cursor key, the control, does not lose focus when the description is read out (i.e. moves cursor to description), so it would be nice if worked the same in reverse.
4. Once a balloon or a button has focus, I cannot use the up cursor to arrow back into the Scene Summary. I get stuck on the Sweater description.

These issues are likely related to the new two-step balloon interaction (#133).

**Comment by terracoda** [comment]
@jessegreenberg, FYI, I made a couple updates to clarify wording in the above comment.

**Comment by jessegreenberg** [comment]
> 1. The label of the Grab Balloon button is not read out, and the button (the balloon) does not gain visual focus, whereas this is how the other buttons work.

Good catch @terracoda, yes likely due to changes related to #133.  Seems to be introduced as of version reader.3.

> 2. The Yellow balloon description (before interaction), contains no position nor charge information. It only contains the navigation cue which is actually inoperable when the balloon is not grabbed.

Are there other descriptions in the design document for Perspective 1?  Before the interaction, there is the 'grab balloon' label, and the navigation help description.  It looks like columns for position and charge info are blank.

> 3. When using the up cursor key (i.e. moving backwards), it would make sense to have the control gain focus when the user cursors to the control's description (is that possible?). Using the down cursor key, the control, does not lose focus when the description is read out (i.e. moves cursor to description), so it would be nice if worked the same in reverse.

Most screen readers do not do this, so I didn't implement it in the prototype.  If we did this, the user would hear the focused control, not its description.  I think if the user presses 'up arrow' to find a description they would be expecting the description not the element itself.  Like other screen readers, the prototype sets focus if the cursor finds something that is focusable.

> 4. Once a balloon or a button has focus, I cannot use the up cursor to arrow back into the Scene Summary. I get stuck on the Sweater description.

I can't reproduce this issue on Chrome or Safari.  Which platform were you testing on?

**Comment by jessegreenberg** [comment]
Continued work in #133 will likely fix point 1 in the initial issue comment.

**Comment by terracoda** [comment]
@jessegreenberg regarding "getting stuck on the Sweater description while using the up cursor key", this only happens while using Safari 9.1.1 (El Capitan, Mac Air).

**Comment by jessegreenberg** [comment]
Thanks @terracoda, I will try to reproduce on that platform.

**Comment by terracoda** [comment]
@jessegreenberg, sorry, regarding the blanks columns, I see the documentation is unclear. P1 was originally focused  and grabbed but not yet moving. I need to make this more clear.
> Are there other descriptions in the design document for Perspective 1? Before the interaction, there is the 'grab balloon' label, and the navigation help description. It looks like columns for position and charge info are blank.

I might have removed the content thinking that there would be no balloon at all until it was grabbed.

**Comment by terracoda** [comment]
@jessegreenberg, regarding second part of point 3, 
> Using the down cursor key, the control, does not lose focus when the description is read out (i.e. moves cursor to description)
What I meant to say is that the control remains operable, even after the user cursors down to the description. And this does not happen in reverse, obviously, because I am going backwards.

So, I agree with you:
> Most screen readers do not do this, so I didn't implement it in the prototype. If we did this, the user would hear the focused control, not its description. I think if the user presses 'up arrow' to find a description they would be expecting the description not the element itself. Like other screen readers, the prototype sets focus if the cursor finds something that is focusable.

I was definitely projecting too much of what I see visually and hear onto what I think what a screen reader will understand from listening alone. Number 3 is not a bug :-)

**Comment by jessegreenberg** [comment]
Issues with the balloon descriptions are in progress via #154 and the interaction is being tracked in #133.  Points 1 and 2 above have been fixed.  If point 4 above resurfaces, we can create a new issue in scenery to track it.  Closing.
